source/csharp/Camera/Basic/CaptureViaGenICam/CaptureViaGenICam.cs

```diff
@@ -0,0 +1,128 @@
+                                    0,
+                                    "progressive",
+                                    -1,
+                                    "default",
+                                    -1,
+                                    "false",
+                                    "default",
+                                    zividDevice,
+                                    0,
+                                    0,
+                                    out framegrabber);
+
+            Console.WriteLine("Configuring 3D-settings");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "create_objectmodel3d", "enable");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "add_objectmodel3d_overlay_attrib", "enable");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "AcquisitionMode", "SingleFrame");
+
+            Console.WriteLine("Configuring camera settings");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "Aperture", 5.66);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ExposureTime", 8333);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "Gain", 1);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ProcessingFiltersOutlierRemovalEnabled", 1);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ProcessingFiltersOutlierRemovalThreshold", 5);
+
+            Console.WriteLine("Capturing frame");
+            var region = new HalconDotNet.HObject();
+            var contours = new HalconDotNet.HObject();
+            var data = new HalconDotNet.HTuple();
+            var frame = new HalconDotNet.HObject();
+            HalconDotNet.HOperatorSet.GrabData(out frame, out region, out contours, framegrabber, out data);
+
+            var x = frame.SelectObj(1);
+            var y = frame.SelectObj(2);
+            var z = frame.SelectObj(3);
+            var snr = frame.SelectObj(4);
+            var rgb = frame.SelectObj(5);
+
+            Console.WriteLine("Removing invalid 3D points (zeroes)");
+            var reducedRegion = new HalconDotNet.HObject();
+            var zReduced = new HalconDotNet.HObject();
+            HalconDotNet.HOperatorSet.Threshold(z, out reducedRegion, 0.0001, 10000);
+            HalconDotNet.HOperatorSet.ReduceDomain(z, reducedRegion, out zReduced);
+
+            Console.WriteLine("Constructing ObjetModel3D based on XYZ data");
+            var objectModel3D = new HalconDotNet.HTuple();
+            HalconDotNet.HOperatorSet.XyzToObjectModel3d(x, y, zReduced, out objectModel3D);
+
+            Console.WriteLine("Adding RGB to ObjectModel3D");
+            SetColorsInObjectModel3D(objectModel3D, rgb, zReduced);
+
+            string fileName = "Zivid3D.ply";
+            Console.WriteLine("Saving point cloud to: " + fileName);
+            SaveHalconPointCloud(objectModel3D, fileName);
```

Comment:
check what variable name convention we are using in C# / C++ samples, especially for ply. I think we use pointCloud. Let's make it consistent.

Reply:
It does not say anything about the fileName in our internal KB, but from the sample we use pointCloudFile. So switching to this